lib/eex/lib/eex.ex

```diff
@@ -272,6 +272,35 @@ defmodule EEx do
     do_eval(compiled, bindings, options)
   end
 
+  @doc """
+  Tokenize the given contents according to the given options.
+
+  ## options
+
+  * `line` - An interger to start as line. Default is 1.
+  * `column` - An interger to start as line. Default is 1.
+  * `indentation` - An integer that indicates the indentation. Default is 0.
+  * `trim` - Tells the tokenier to either trim the content or not. Default is false.
+
+  ## Examples
+
+      iex> EEx.tokenize('foo', line: 1, column: 1)
+      {:ok, [{:text, 'foo', %{column: 1, line: 1}}, {:eof, %{column: 4, line: 1}}]}
+
+  """
+  @spec tokenize(contents :: binary | charlist, opts :: keyword) ::
+          {:ok, EEx.Tokenizer.token()} | {:error, String.t(), EEx.Tokenizer.metadata()}
+  def tokenize(contents, opts) do
```

Comment:
```suggestion
  def tokenize(contents, opts \\ []) do
```

Maybe make it optional?

Reply:
Yes!

let’s also document the tokens by bringing the specs from EEx.Tokenizer to this module!